pkg/artifactory/resource/repository/local/resource_artifactory_local_terraform_module_repository.go

```diff
@@ -0,0 +1,50 @@
+package local
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/jfrog/terraform-provider-artifactory/v6/pkg/artifactory/resource/repository"
+	"github.com/jfrog/terraform-provider-shared/util"
+	"github.com/jfrog/terraform-provider-shared/validator"
+)
+
+var terraformModuleLocalSchema = util.MergeSchema(
+	BaseLocalRepoSchema,
+	map[string]*schema.Schema{
+		"registry_type": {
+			Type:     schema.TypeString,
+			Optional: true,
+			Default:  "module",
+			Description: "The Terraform registry in Artifactory allows you to create dedicated repositories for the " +
+				"unique Terraform component Module.",
+			ValidateDiagFunc: validator.StringInSlice(true, "module"),
+		},
+	},
+	repository.RepoLayoutRefSchema("local", "terraform_module"),
+)
+
+func ResourceArtifactoryLocalTerraformModuleRepository() *schema.Resource {
+
+	type TerraformLocalRepo struct {
```

Comment:
All the code in this func is identical to the other 'provider' repo. Please move this to a shared package (inside `repository.go` is acceptable.) as a func so we don't have 2 copies.

Reply:
I'm moving it to `local.go`, because it's specific to only local repositories.